### DISCUSSION TITLE
Example Serial-Monitor use with PicoProbe and PlatformIO in VsCode

### DISCUSSION THREAD

**Discussion by TA1DB** [earlephilhower/arduino-pico#1220] [Q&A] [answered]
Is there an example of serial monitor setup and use of arduino-pico with picoprobe and platformio in Vscode? I can upload and debug without any difficulty but I can't use the serial monitor ( UART ). I can use serial monitor of wizio-pico + platformio + picoprobe setup and with openocd-gdb CMakeList.txt configuration (without platformio) as explained  in the getting started manual but I can do it for arduino-pico with picoprobe and platformio in Vscode. 
In another words, I can't use Serial.printf() with picoprobe + platformio .. am I missing something?

**Comment by joba-1** [comment, marked as answer]
Sorry, just a me too. Looks like we are in the same boat. Any progress?

What I did:
On opensuse linux with VS Code and PlatformIO I can connect the pico-w with USB and see Serial.print() output via the matching tty port (env:picow) while the led blinks.
If I use a picoprobe and no usb connected to the target probe I can debug but I cannot see Serial.print() output (env:picow_debug).
I connected a logic analyzer to the target pins 0 and 1 and see just a permanent high level. So it looks like it is not a wiring problem and not a picoprobe problem but a firmware problem on the target.

This is the platformio.ini
```
[platformio]
default_envs = picow_debug

[env]
framework = arduino
monitor_speed = 115200

[env:picow_debug]
platform = https://github.com/maxgerhardt/platform-raspberrypi.git
board_build.filesystem_size = 0.5m
board = rpipicow
upload_protocol = cmsis-dap
debug_tool = cmsis-dap
debug_speed = 5000
monitor_port = /dev/ttyUSB_picoprobe

[env:picow]
platform = https://github.com/maxgerhardt/platform-raspberrypi.git
board = pico
upload_port = /dev/ttyUSB_picow1
upload_protocol = picotool
monitor_port = /dev/ttyUSB_picow1
```

This is my test scetch

```
#include <Arduino.h>

#define LED_PIN 9
#define TOGGLE_DELAY 200

void setup() {
  Serial.begin(115200);
  pinMode(LED_PIN, OUTPUT);
  Serial.println("\nSetup done");
}

void loop() {
  const static uint32_t interval = TOGGLE_DELAY;
  static uint32_t prev_ms = 0;
  uint32_t now = millis();
  if (now - prev_ms > interval)
  {
    prev_ms = now;
    digitalWrite(LED_PIN, digitalRead(LED_PIN) ? LOW : HIGH);
    Serial.println("Toggle LED");
  }
}
```

**Reply by earlephilhower** [reply]
> ...I cannot see Serial.print() output (env:picow_debug).  I connected a logic analyzer to the target pins 0 and 1 and see just a permanent high level...

`Serial.print()` goes to USB.  `Serial1.print()` is to UART0,if you really want UART serial output.  Also, of course, check your boards default serial pins since there are multiple options.

**Reply by TA1DB** [reply]
It's OK now thanks a lot ! @joba-1 thanks to you, too !

**Comment by joba-1** [comment]
So I mixed up the numbering. Somehow I was lead to believe UART1->Serial1 and UART0/USB->Serial.
Using Serial1.print()s actually work.
Thank you!
